opentelemetry-sdk/src/logs/record.rs

```diff
@@ -31,13 +34,35 @@ pub struct LogRecord {
     pub attributes: Option<Vec<(Key, AnyValue)>>,
 }
 
+impl Default for LogRecord {
+    fn default() -> Self {
+        LogRecord {
+            event_name: None,
+            timestamp: None,
+            observed_timestamp: SystemTime::now(),
+            trace_context: None,
+            severity_text: None,
+            severity_number: None,
+            body: None,
+            attributes: None,
+        }
+    }
+}
+
 impl opentelemetry::logs::LogRecord for LogRecord {
+    fn set_event_name<T>(&mut self, name: T)
+    where
+        T: Into<Cow<'static, str>>,
+    {
+        self.event_name = Some(name.into());
+    }
+
     fn set_timestamp(&mut self, timestamp: SystemTime) {
         self.timestamp = Some(timestamp);
     }
 
     fn set_observed_timestamp(&mut self, timestamp: SystemTime) {
-        self.observed_timestamp = Some(timestamp);
+        self.observed_timestamp = timestamp;
```

Comment:
I think we shouldn't allow users to change observed timestamp as that only depends on when the SDK collected the event.

Reply:
Agree. Let's wait for more comment if all agree :)

Reply:
We need to make the `observed_timestamp` field private then. I think that makes sense

Reply:
Going through the [specs ](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#emit-a-logrecord) it say's:

```
Emit a LogRecord
The effect of calling this API is to emit a LogRecord to the processing pipeline.

The API MUST accept the following parameters:
...
[Observed Timestamp] If unspecified the implementation SHOULD set it equal to the current time.
```

My interpretation is that the API should provide interface to set the `observed timestamp`, and use current time if not set.

Reply:
Ok, I have reverted `observed_timestamp` type back to Optional, and kept it's setter function for the appenders. If the timestamp is not set, the SDK will set it to it's current time.

Reply:
Ah if the spec says so then we should probably leave it as it is